doc/developer/design/20231027_refresh_every_mv.md

```diff
@@ -0,0 +1,164 @@
+
+ODW use cases typically focus on the most recent input data (hot data). However, there is often a significantly larger amount of older data lying around (cold data). The cold data is often of the same kind as the hot data, and therefore we postulate that users will often want to perform the same or similar computations on these. In this case, it would be convenient for users to simply use Materialize for both hot and cold computations, and not complicate their architectures by implementing the same computations in both Materialize and a different, traditional data warehouse.
+
+The problem is that Materialize is currently not so cost-effective for running a materialized view on cold data, because
+- there is significantly more cold data than hot data, so a much larger compute instance is needed to keep compute state in memory, and
+- keeping results always fresh is not so valuable for cold data.
+
+This currently prevents us from capturing these cold use cases, even though they feel very close to our hot, core ODW use cases.
+
+## Success Criteria
+
+We should give users a way to trade off freshness vs. cost.
+
+Note that other systems also expose similar knobs. Making this trade-off tunable by users is an important point also in the [Napa paper](http://www.vldb.org/pvldb/vol14/p2986-sankaranarayanan.pdf). Snowflake's dynamic tables allow the user to configure a ["target lag"](https://docs.snowflake.com/en/user-guide/dynamic-tables-refresh#understanding-target-lag).
+
+1. Importantly, our solution shouldn't complicate the user's architecture. For example, the user shouldn't need to externally schedule jobs that perform an array of non-trivial tasks (and manage their possible failures), such as managing multistep table updates (e.g., first delete old results than insert new results) and turning replicas on and off. This is a crucial criterion, because if we can only propose a complicated solution, then users won't be better off with our solution than with using a different system for the cold data.
+
+2. Another critical requirement is that our solution should achieve a significant cost reduction. This is because the cold data is often significantly larger than the hot data. For example, at one important user, the hot data is from the last 14 days, while the cold data is the same data but from the last 5 years.
+
+3. The slowly updated results should be queryable together with other, normal objects. This is important because sometimes we might want to join the cold results with other objects. Also, one important user needs a unified view of the hot and cold results, so we should be able to union them.
+
+4. We shouldn't hold back Persist compaction of inputs to the time of the last refresh. This is to avoid potentially interfering with other parts of the system.
```

Comment:
Not just `persist` right, but other components as well? Indexes are the most scary ones, but I do agree that we also shouldn't hold back persist compaction.